Jenkinsfile

```diff
@@ -0,0 +1,305 @@
+pipeline {
+    parameters {
+        booleanParam defaultValue: false,
+                description: 'Whether to upload the packages in playground repositories',
+                name: 'PLAYGROUND'
+    }
+    options {
+        skipDefaultCheckout()
+        buildDiscarder(logRotator(numToKeepStr: '5'))
+        timeout(time: 3, unit: 'HOURS')
+    }
+    agent {
+        node {
+            label 'base-agent-v1'
+        }
+    }
+    environment {
+        NETWORK_OPTS = '--network ci_agent'
+        ARTIFACTORY_ACCESS=credentials('artifactory-jenkins-gradle-properties-splitted')
+    }
+    stages {
+        stage('Checkout') {
+            steps {
+                checkout scm
+            }
+        }
+        stage('Stash') {
+            steps {
+                sh 'mkdir staging'
+                sh 'cp -r package conf staging'
+                stash includes: 'staging', name: 'staging'
+            }
+        }
+        stage('Package') {
+            parallel {
+                stage('Ubuntu 20.04') {
+                    agent {
+                        node {
+                            label 'yap-agent-ubuntu-20.04-v2'
+                        }
+                    }
+                    steps {
+                        unstash 'staging'
+                        sh 'cp -r staging /tmp'
+                        sh 'sudo yap build ubuntu-focal /tmp/staging/package'
+                        stash includes: 'artifacts/',
+                                name: 'artifacts-ubuntu-focal'
+                    }
+                    post {
+                        always {
+                            archiveArtifacts artifacts: 'artifacts/*focal*.deb',
+                                    fingerprint: true
+                        }
+                    }
+                }
+                stage('Ubuntu 22.04') {
```

Comment:
you can also remove ubuntu22 and rhel9 stages having only an "Ubuntu" and RHEL stage. You can find examples of what I'm saying if you search for "yap build ubuntu" and "yap build rocky" with no focal or 8/9 specification.

The only two generated packages will be then uploaded on the related repos. Maybe  the pending-setups repo can give you an idea.

The only exception would be carbonio-core package (only because of different required dependencies)